flytekit/annotated/workflow.py

```diff
@@ -422,35 +438,17 @@ def get_registerable_entity(self) -> _SdkWorkflow:
         return self._registerable_entity
 
 
-def workflow(
-    _workflow_function=None,
-    reference: Optional[WorkflowReference] = None,
-    failure_policy: Optional[WorkflowFailurePolicy] = None,
-    interruptible: Optional[bool] = False,
-):
+def reference_workflow(
+    project: str, domain: str, name: str, version: str,
+) -> Callable[[Callable[..., Any]], ReferenceWorkflow]:
     """
-    :param reference: Pass a WorkflowReference object here if you want to make this a pointer to an existing workflow
-    :param failure_policy: Use the options in flytekit.WorkflowFailurePolicy
-    :param interruptible: Whether or not tasks launched from this workflow are by default interruptible
+    A reference workflow is a pointer to a workflow that already exists on your Flyte installation. This
+    object will not initiate a network call to Admin, which is why the user is asked to provide the expected interface.
+    If at registration time the interface provided causes an issue with compilation, an error will be returned.
     """
 
-    # Unlike for tasks, where we can determine the entire structure of the task by looking at the function's signature,
-    # workflows need to have the body of the function itself run at module-load time. This is because the body of the
-    # workflow is what expresses the workflow structure.
-    def wrapper(fn):
-        if reference:
-            workflow_instance = ReferenceWorkflow.create_from_function(fn, reference=reference)
-            return workflow_instance
-
-        workflow_metadata = WorkflowMetadata(on_failure=failure_policy or WorkflowFailurePolicy.FAIL_IMMEDIATELY)
-
-        workflow_metadata_defaults = WorkflowMetadataDefaults(interruptible)
-
-        workflow_instance = Workflow(fn, metadata=workflow_metadata, default_metadata=workflow_metadata_defaults)
-        workflow_instance.compile()
-        return workflow_instance
+    def wrapper(fn) -> ReferenceWorkflow:
+        interface = transform_signature_to_interface(inspect.signature(fn))
+        return ReferenceWorkflow(project, domain, name, version, interface.inputs, interface.outputs)
 
-    if _workflow_function:
-        return wrapper(_workflow_function)
-    else:
-        return wrapper
+    return wrapper
```

Comment:
same question about whether we need to raise an error if the user forgets to fill in the `(...)`

Reply:
`project: str, domain: str, name: str, version: str,` are all mandatory now without default value, so there should be a hard error if only `@reference_workflow()` is used.

Reply:
`@reference_workflow` only will also result error like `missing required positional arguments`

Reply:
good